test/extended/prometheus/prometheus.go

```diff
@@ -184,6 +184,17 @@ var _ = g.Describe("[Feature:Prometheus][Conformance] Prometheus", func() {
 			}
 			runQueries(tests, oc, ns, execPod.Name, url, bearerToken)
 		})
+		g.It("shouldn't experience cardinality explosion", func() {
+			oc.SetupProject()
+			ns := oc.Namespace()
+			execPod := exutil.CreateCentosExecPodOrFail(oc.AdminKubeClient(), ns, "execpod", nil)
+			defer func() { oc.AdminKubeClient().CoreV1().Pods(ns).Delete(execPod.Name, metav1.NewDeleteOptions(1)) }()
+
+			tests := map[string]bool{
+				`rate(scrape_series_added[10m]) > 100`: false,
```

Comment:
I'm pretty sure kube-state-metrics will fail this due to pod churn.

Reply:
Maybe we can measure kube-state-metrics (and potentially all current targets in openshift) and tweak the setting?

Reply:
We can exclude ksm from cardinality check until we have some better way of doing it. wdyt?

Reply:
Using the rate function is a bit odd here no? I feel we should rather use the increase function.

Reply:
You are right, especially that this is applying `rate()` to counter.

I checked `increase(scrape_series_added[30m])` and it seems that worst offenders are KSM and both apiservers (kube + openshift). Additionally with such query it makes much more sense to have separate queries for KSM as well as apiservers. As for time range I think 30m should be fine.

This time graphs are from prometheus UI as Console UI shows completely different results in my cluster :man_shrugging: 
![Screenshot_20191106_122106](https://user-images.githubusercontent.com/3531758/68294223-f17c4d00-008f-11ea-837c-b1075b957b8a.png)

Tail stabilizes at less than 200 and the distribution of the beginning is as follows:
- job=KSM < 15.5k
- job=apiserver < 6.5k
- job=api < 2k
- others < 400

With that I think the following queries would be appropriate to check for cardinality explosion (fail test if any is true):
```
increase(scrape_series_added{job="kube-state-metrics"}[30m]) > 20000
increase(scrape_series_added{job="apiserver"}[30m]) > 8000
increase(scrape_series_added{job="api"}[30m]) > 3000
increase(scrape_series_added{job!~"kube-state-metrics|apiserver|api"}[30m]) > 1000
```

Alternative would be to run `sum(increase(scrape_series_added[30m])) by (job)` which will give us sums so we wouldn't be relying on load distribution. I am not sure of this approach, but I think it might be a better one. For this graph is the following:
![Screenshot_20191106_123929](https://user-images.githubusercontent.com/3531758/68295342-7cf6dd80-0092-11ea-9f32-16f841975c5b.png)

In this case, only apiservers are the worst offenders with values of ~20.5k and ~4.5k for apiserver and api respectively. Everything else is less than 400. For this query I would suggest following tests:
```
sum(increase(scrape_series_added{job="apiserver"}[30m])) by (job) > 25000
sum(increase(scrape_series_added{job="api"}[30m])) by (job) > 7000
sum(increase(scrape_series_added{job!~"api|apiserver"}[30m])) by (job) > 1000
```

@brancz @LiliC wdyt?

Reply:
see https://github.com/openshift/origin/pull/24074#discussion_r342968675
`increase`/`rate` shouldn't be used IMHO since `scrape_series_added` isn't a counter.